codex-cli/src/cli.tsx

```diff
@@ -267,8 +267,14 @@ const imagePaths = cli.flags.image;
 const provider = cli.flags.provider ?? config.provider ?? "openai";
 const apiKey = getApiKey(provider);
 
-// Set of providers that don't require API keys
-const NO_API_KEY_REQUIRED = new Set(["ollama"]);
+// Set of providers that don't require API keys, "ollama" for backwards compat
+// and those providers which don't provide `envKey` in configuration are exempt
+const NO_API_KEY_REQUIRED = new Set([
```

Comment:
This map does not seem necessary. Below, just replace:

```
!NO_API_KEY_REQUIRED.has(provider.toLowerCase())
```

with something like:

```
config.providers?[provider]?.envKey != null
```